android/source/VideoLocker/src/org/edx/mobile/view/custom/ESpinner.java

```diff
@@ -0,0 +1,129 @@
+package org.edx.mobile.view.custom;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.view.Gravity;
+import android.view.View;
+
+import org.edx.mobile.R;
+import org.edx.mobile.util.UiUtil;
+
+import java.util.List;
+
+/**
+ * Created by rohan on 2/16/15.
+ */
+public class ESpinner<T> extends ETextView implements View.OnClickListener,
```

Comment:
the name of this is kind of surprising. I was expecting it to be a subclass of ``Spinner``, not of a text view. Maybe we can name it so it's clear it's associated with a text field?